plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1410,6 +1413,22 @@ public Optional<ConnectorOutputMetadata> finishInsert(
         return Optional.empty();
     }
 
+    private static List<String> getWritePartitionColumnNames(List<String> originalPartitionColumns, List<DeltaLakeColumnHandle> dataColumns)
+    {
+        return originalPartitionColumns.stream()
+                .map(columnName -> {
+                    DeltaLakeColumnHandle dataColumn = dataColumns.stream()
+                            .filter(column -> columnName.equalsIgnoreCase(column.getName()))
+                            .collect(onlyElement());
+                    // Note: during writes we want to preserve original case of partition columns, if the column's name is not differ of column's physical name
+                    if (dataColumn.getPhysicalName().equalsIgnoreCase(columnName)) {
```

Comment:
Does this apply only for column mapping `NONE` ?

Reply:
Agreed, if the behavior is very different for `name` mapping I'd rewrite this method to only do the name mapping and call it `getPartitionColumnsForNameMapping`

Reply:
In the case of a `none` mapping you're just returning the input `handle.getMetadataEntry().getOriginalPartitionColumns()` right?